src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,55 @@
+package org.opentripplanner.middleware.utils;
+
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.Leg;
+
+import java.util.Collection;
+import java.util.List;
+
+public class ItineraryUtils {
+    private static final Collection<String> OTP_TRANSIT_MODES = List.of("TRAM", "BUS", "SUBWAY", "FERRY", "RAIL", "GONDOLA");
```

Comment:
These possible modes are missing: `CABLE_CAR`, `FUNICULAR`, `AIRPLANE`

Reply:
We'll have to add them to OTP-UI too!

Reply:
In https://github.com/ibi-group/otp-middleware/pull/91/commits/8cc7dd929604fe9ba216dcfe41d5ed59161da22c, I ended up using fields `leg.transitLeg`, `leg.rentedVehicle` and `leg.hailedCar` that are populated by OTP.